src/hypervisor.c

```diff
@@ -128,13 +168,43 @@ cc_oci_append_network_args(struct cc_oci_config *config,
 		g_ptr_array_add(additional_args, g_strdup("-net\nnone\n"));
 	} else {
 		for (guint index = 0; index < g_slist_length(config->net.interfaces); index++) {
-			netdev_params = cc_oci_expand_netdev_cmdline(config, index);
-			net_device_params = cc_oci_expand_net_device_cmdline(config, index);
 
-			g_ptr_array_add(additional_args, g_strdup("-netdev"));
-			g_ptr_array_add(additional_args, netdev_params);
-			g_ptr_array_add(additional_args, g_strdup("-device"));
-			g_ptr_array_add(additional_args, net_device_params);
+			if_cfg = (struct cc_oci_net_if_cfg *)
+				g_slist_nth_data(config->net.interfaces, index);
+
+
+			/*
+			 * OVS-DPDK is a special case which requires additional
+			 * parameters
+			 */
+			if (is_interface_ovs(if_cfg)) {
+
+				g_ptr_array_add(additional_args, g_strdup("-chardev"));
+				ovsdpdk_params = cc_oci_expand_ovsdpdk_chardev_params(if_cfg);
+				g_ptr_array_add(additional_args, ovsdpdk_params);
+
+				g_ptr_array_add(additional_args, g_strdup("-netdev"));
+				g_ptr_array_add(additional_args, g_strdup(DPDK_CMDLINE_NETD));
+
+				g_ptr_array_add(additional_args, g_strdup("-device"));
+				g_ptr_array_add(additional_args, g_strdup_printf(DPDK_CMDLINE_DEV,if_cfg->mac_address));
+
+				g_ptr_array_add(additional_args, g_strdup("-object"));
+				g_ptr_array_add(additional_args, g_strdup(DPDK_CMDLINE_OBJ));
+
+				g_ptr_array_add(additional_args, g_strdup("-numa"));
```

Comment:
This should be moved outside this if and applied if there are any ovs interfaces. That will allow us to support multiple ovs interfaces assigned to the same container.